gcloud/storage/bucket.py

```diff
@@ -50,7 +50,7 @@ def get_items_from_response(self, response):
         """
         self.prefixes = tuple(response.get('prefixes', ()))
         for item in response.get('items', []):
-            yield Blob(properties=item, bucket=self.bucket)
+            yield Blob(None, properties=item, bucket=self.bucket)
```

Comment:
That was the previous assumption.

FWIW I'm not sure how I feel about having an explicit `None` for `Blob`s with no name. Maybe using `*args` and allowing length 0 or 1 is appropriate or just have a kwarg that defaults to `None`.

This is brings into question the mutability of a `Blob` and potential existence of something like `Key.completed_key`.